requestmatch/matcher_test.go

```diff
@@ -1093,6 +1090,44 @@ func TestFreeWildcardParamWithSlash(t *testing.T) {
 	}
 }
 
+func TestHeaderMatchCaseInsensitiveTo(t *testing.T) {
+	routes, err := eskip.Parse(`Header("some-header", "some-value") -> "https://example.org"`)
+	if err != nil {
+		t.Error(err)
+	}
+
+	rd := &def{routes[0]}
+
+	m, errs := Make([]Definition{rd}, false)
+	if len(errs) != 0 {
+		t.Error("failed to make matcher")
+	}
+
+	r, _ := m.Match(&http.Request{URL: &url.URL{}, Header: http.Header{"Some-Header": []string{"some-value"}}})
+	if r == nil {
+		t.Error("failed to match header, case insensitive, to")
+	}
+}
+
+func TestHeaderMatchCaseInsensitiveFrom(t *testing.T) {
+	routes, err := eskip.Parse(`Header("Some-Header", "some-value") -> "https://example.org"`)
+	if err != nil {
+		t.Error(err)
+	}
+
+	rd := &def{routes[0]}
+
+	m, errs := Make([]Definition{rd}, false)
+	if len(errs) != 0 {
+		t.Error("failed to make matcher")
+	}
+
+	r, _ := m.Match(&http.Request{URL: &url.URL{}, Header: http.Header{"some-header": []string{"some-value"}}})
```

Comment:
A lot of repetitive boilerplate here.... Couldn't this be pulled into a func that takes an eskip string and a request (and maybe a flag to indicate whether there should be a route or not)?